### PR TITLE
Milestone 5 # Add Interaction

### PR DESCRIPTION
**Pull request by gscarv13**
### This pull request contain the following:
- Added states to App component
- Share the states through out the components
- Fix the `calculate` module to accept multiple numbers

### CODE REVIEW

File: src/components/ButtonPanel.js [hunk 1]
```diff
@@ -10,20 +11,24 @@ const GroupButtons = [
...
     <div key={`group${groupID}`}>
-      { group.map((btn, btnID) => <Button key={`btn${btnID}`} name={btn} />) }
+      { group.map((btn, btnID) => <Button key={`btn${btnID}`} name={btn} clickHandler={handler} />) }
     </div>
   ));
 }
 
-function ButtonPanel() {
+function ButtonPanel(props) {
+  const { clickHandler } = props;
```
Comment (rindrajosia) [inline src/components/ButtonPanel.js]: it implements `handleClick` function

File: src/components/Button.js [hunk 2]
```diff
@@ -2,16 +2,17 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 function Button(props) {
-  const { name } = props;
+  const { name, clickHandler } = props;
   return (
-    <button type="button">
+    <button type="button" onClick={() => clickHandler(name)}>
```
Comment (rindrajosia) [inline src/components/Button.js]: it attaches `handleClick` to `onClick` event on Button

File: src/components/Button.js [hunk 3]
```diff
@@ -2,16 +2,17 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 function Button(props) {
```
Comment (rindrajosia) [inline src/components/Button.js]: It will be nice if you use the arrow function

File: src/components/ButtonPanel.js [hunk 4]
```diff
@@ -10,20 +11,24 @@ const GroupButtons = [
   ['0', '.', '='],
 ];
 
-function displayButtons(buttons) {
+function displayButtons(buttons, handler) {
```
Comment (rindrajosia) [inline src/components/ButtonPanel.js]: It will be nice if you use the arrow function

File: src/components/ButtonPanel.js [hunk 5]
```diff
@@ -10,20 +11,24 @@ const GroupButtons = [
...
   return buttons.map((group, groupID) => (
     <div key={`group${groupID}`}>
-      { group.map((btn, btnID) => <Button key={`btn${btnID}`} name={btn} />) }
+      { group.map((btn, btnID) => <Button key={`btn${btnID}`} name={btn} clickHandler={handler} />) }
     </div>
   ));
 }
 
-function ButtonPanel() {
+function ButtonPanel(props) {
```
Comment (rindrajosia) [inline src/components/ButtonPanel.js]: It will be nice if you use the arrow function

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- App component is converted into stateful component :+1: 
- it passes `handleClick` as a `clickHandler` property to `ButtonPanel` :100: 
- it passes the `total` or `next` value from state to `Display` :1st_place_medal: 

To improve :factory_worker: [REQUIRED]:

- [x] `ButtonPanel` component should implement `handleClick` function ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/panel/implementHandleClick.png))
- [x] `Button` component should attache `handleClick` to `onClick` event on Button
- [x] As you are using ES6 syntax, please use the arrow function instead of the function keyword.
- [x] I suggest that you update your live demo

Nice to have [OPTIONAL] :

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Comment by gscarv13** [comment]
### Thanks for your review @rindrajosia !
As suggested I updated the functions and renamed the `clickHandler` and `handler` into handleClick

**Review by enaburekhan — APPROVED** [review]
# Status: Approve:heavy_check_mark:
**Hi Silva!
Congratulations:clap::clap:. Thank you for implementing the changes requested by the last reviewer.
Move on with the next project:walking:.**

**Optional**

- [x] When I divide a number by zero, I get this [error](https://imgur.com/a/GqfG42T). It will be nice to fix this before merging the PR.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Pull request merged**